### PR TITLE
kill all unused variables in apps/src

### PR DESCRIPTION
**Pull request by pcardune**
hope this doesn't break anything 😇

### CODE REVIEW

File: apps/src/StudioApp.js [hunk 1]
```diff
@@ -2019,12 +2017,6 @@ StudioApp.prototype.handleEditCode_ = function (config) {
     return;
   }
 
-  var displayMessage, examplePrograms, messageElement, onChange, startingText;
-
-  // Ensure global ace variable is the same as window.ace
-  // (important because they can be different in our test environment)
-  ace = window.ace;
```
Comment (islemaster) [inline apps/src/StudioApp.js]: This looks important.  Have you tested around this?

File: apps/src/dropletUtils.js [hunk 2]
```diff
@@ -32,10 +32,8 @@ var _ = require('lodash');
  * @property {Object} categories configuration of categories within which to place blocks
  */
 
-var COLOR_PINK = '#F57AC6';
 var COLOR_PURPLE = '#BB77C7';
 var COLOR_GREEN = '#68D995';
-var COLOR_LIGHT_GREEN = '#D3E965';
```
Comment (islemaster) [inline apps/src/dropletUtils.js:38]: I assume this set of colors is here for some reason?  Might be worth tracking down / annotating an exception for.

File: apps/src/gamelab/PropTypes.js [hunk 3]
```diff
@@ -131,16 +130,6 @@ export const AnimationList = React.PropTypes.shape({
 });
 
 /**
- * @typedef {Object} SerializedAnimationList
- * @property {AnimationKey[]} orderedKeys - Animations in project order
- * @property {Object.<AnimationKey, SerializedAnimationProps>} propsByKey
- */
```
Comment (islemaster) [inline apps/src/gamelab/PropTypes.js]: Please leave the `@typedef` comment in here.

File: apps/src/gamelab/dropletConfig.js [hunk 4]
```diff
@@ -9,12 +8,7 @@ var COLOR_LIGHT_GREEN = '#D3E965';
 var COLOR_RED = '#F78183';
 var COLOR_CYAN = '#4DD0E1';
 var COLOR_YELLOW = '#FFF176';
-var COLOR_PINK = '#F57AC6';
-var COLOR_PURPLE = '#BB77C7';
-var COLOR_GREEN = '#68D995';
-var COLOR_WHITE = '#FFFFFF';
 var COLOR_BLUE = '#64B5F6';
-var COLOR_ORANGE = '#FFB74D';
```
Comment (islemaster) [inline apps/src/gamelab/dropletConfig.js:17]: I bet these are duplicated _everywhere_ - we should centralize them in a future PR.

File: apps/src/.eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  "rules": {
+    "no-unused-vars": ['error', {"args": 'none'}],
+  }
+};
```
Comment (islemaster) [inline apps/src/.eslintrc.js]: Can we push this rule all the way up to `[cdo]/.eslintrc.js` yet?

File: apps/src/code-studio/.eslintrc.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  "rules": {
+    "no-unused-vars": ['error', {"args": 'none'}],
+  }
+};
```
Comment (islemaster) [inline apps/src/code-studio/.eslintrc.js]: This file is redundant with `[cdo]/apps/src/.eslintrc.js` and should be removed.

Comment (islemaster) [reply apps/src/code-studio/.eslintrc.js]: Could also remove the one you just added for src/applab.

File: apps/src/StudioApp.js (hunk 1 above)
Comment (pcardune) [reply apps/src/StudioApp.js]: I don't understand under what circumstances this code actually does anything. `ace` isn't referenced anywhere in this file, and even if this line is considered to be setting the global `ace` variable, wouldn't this just be equivalent to `window.ace = window.ace`?

File: apps/src/.eslintrc.js (hunk 5 above)
Comment (pcardune) [reply apps/src/.eslintrc.js]: nope. I haven't tackled the stuff in `apps/src/test` yet.

File: apps/src/StudioApp.js (hunk 1 above)
Comment (islemaster) [reply apps/src/StudioApp.js]: I don't understand it either.  It looks like this comment was added by @cpirich in beca7d0 a long time ago, he may know something about it.  Possible this refers to a time when some of our tests ran on node without phantom?  In any case, if neither of us understand it well I'd like to be cautious about removing it.

**Comment by pcardune** [comment]
Ok, verified that the only tests which are failing now are the rails url routing issues which are already in staging. I'm going to go ahead and merge this.

**Pull request merged**